### ISSUE TITLE
TailTipWidget issues

### ISSUE THREAD

**Issue by krzsal** [jline/jline3#665]
Hi,
When using Windows Command Prompt od Power Shell with TailTipWidget I have following issues:
1. On the start the prompt is reprinted in previous line and cursor is there also. 
![backed_prompt](https://user-images.githubusercontent.com/79090338/114548109-9aed4800-9c5f-11eb-93d0-6f34be722e53.png)
This behaviour is not visible in gitbash(windows) or unix terminal. Without TailTips enabled everything is ok.
![regular_prompt](https://user-images.githubusercontent.com/79090338/114548169-afc9db80-9c5f-11eb-9499-a8554a7f9a72.png)
2. The after-cursor suggestion when very long, the typing that causes new line break makes suggestion to start loosing characters in new line. However hitting onec "backspace" will fix the suggestion.
![long_suggestion_broken](https://user-images.githubusercontent.com/79090338/114548742-734aaf80-9c60-11eb-80d7-845254f5e160.png)
This issue is not visible for xterm terminals.
3. Having only one suggestion, I would expect that it will disappear when I start typing but it looks like that this one suggestion is further somehow parsed.
![parsing](https://user-images.githubusercontent.com/79090338/114548931-b4db5a80-9c60-11eb-8ea3-258f97b19799.png)
![parsing_stays](https://user-images.githubusercontent.com/79090338/114548954-b9077800-9c60-11eb-83f6-287e7b7607dc.png)

Regarding the code for terminal and reader:
```
Terminal terminal = TerminalBuilder.builder().system(true)
                    .streams(System.in, System.out).build();
LineReader reader =
                LineReaderBuilder.builder().terminal(terminal).completer(completer)
                        .option(LineReader.Option.BRACKETED_PASTE, false)
                        .build();
```
and for TailTipWidget:
```
Map<String, CmdDesc> tailTips = new HashMap<>();
Map<String, List<AttributedString>> widgetOpts = new HashMap<>();
for (String cmd : cmdSyntaxMap.keySet()) { 
/* The cmd variable will be e.g. for ping:
    "[-n|--number <number_of_pings>] [-s|--size <bytes>] [-b|--time_between <seconds|milliseconds>] [-t|--timeout <seconds>] [<address>]"
     or for httpdownload:
    "<url> [-t|--timeout <seconds>]"
     so It will put only one argument (String) per command.
*/
    tailTips.put(cmd, new CmdDesc(Arrays.asList(new AttributedString("")), ArgDesc.doArgNames(Arrays.asList(cmdSyntaxMap.get(cmd))), widgetOpts));
}
TailTipWidgets tailtipWidgets = new TailTipWidgets(reader, tailTips, 5, TailTipWidgets.TipType.TAIL_TIP);
tailtipWidgets.enable();
```

Btw. as you can see the example of the syntax in the code above I have tried to do autocompletion for it but I meet few problems. Is it possible to make completer that will know that argument will have some value (e.g. --timeout someValue) and autocompletion/suggestion will continue after the value?
Thanks in advance.

**Comment by krzsal** [comment]
The 1st mentioned issue is probably due to my workaround to https://github.com/jline/jline3/issues/645 , as I am using following code to hide the widget pane:
```
KeyMap<Binding> keyMap = reader.getKeyMaps().get("main");
keyMap.bind(new Reference(Widgets.TAILTIP_PANE), KeyMap.alt("w"));
reader.runMacro(alt('w'));
```
I notced that hitting alt+w will move cursor to previous line sometimes.

**Comment by krzsal** [comment]
Changing terminal type to `windows` solved the issues 1 & 2. I did not have time to check other things but I will try this week.
Thanks.

**Comment by krzsal** [comment]
Using the OptionCompleter I was able to do what I needed based on your suggestions.

Regarding the TailTip issue 3, it seems that in my case the problem is 1st space character. As when I have put a single String as suggestion: `"<url> [-t|--timeout <seconds>]"` the suggestion is always parsed/cut after space character, like I showed in screenshots. The only solution for that string to be treated as a single suggestion was to put 2 spaces before the string.
Is there other way for complex strings to handle it as a single suggestion? Especially as [ArgDesc.doArgNames](https://github.com/jline/jline3/blob/5be0e7c3012f209ecff6229ddecb865cdb6fc7f6/console/src/main/java/org/jline/console/ArgDesc.java#L37) uses List, thus putting only one element into it will not be parsed, that was my understanging.
Thank you.
